# EDA 프로젝트(Youtube) - Numpy& Pandas

## 필요한 패키지 불러오기

In [5]:
import numpy as np
import pandas as pd

## 데이터 불러오기(KR_youtube_trending_data.csv, category_name.csv)
- ‘category_name.csv’는 데이터에 있는 ‘categoryId’에 대해 더 자세히 설명되어 있는 데이터입니다. (reference 링크에 있는 json 파일을 csv로 변환한 형태입니다.)
- KR_youtube_trending_data.csv는 변수명 df로, category_name.csv는 변수명 cat 으로 가져와주세요. (꼭 변수명이 df와 cat일 필요는 없지만 편의를 위해 설정했습니다.)

In [6]:
# 사용할 데이터셋 불러오기
df = pd.read_csv("C:/Users/김지희/DSL/dataset/KR_youtube_trending_data.csv")
cat = pd.read_csv("C:/Users/김지희/DSL/dataset/category_name.csv")

## df에 대해 탐구해보아요. (중간중간에 df 혹은 df.head()를 계속 출력하면서 잘 되고 있는지 확인해주세요~~)

In [7]:
df.head()

video_id                                             title  \
0  uq5LClQN3cE                                       안녕하세요 보겸입니다   
1  I-ZbZCHsHD0                                부락토스의 계획 [총몇명 프리퀄]   
2  9d7jNUjBoss                                   평생 반성하면서 살겠습니다.   
3  3pI_L3-sMVg                              안녕하세요 꽈뚜룹입니다.   
4  zrsBjYukE8s  박진영 (J.Y. Park) When We Disco (Duet with 선미) M/V   

            publishedAt                 channelId       channelTitle  \
0  2020-08-09T09:32:48Z  UCu9BCtGIEr73LXZsKmoujKw              보겸 BK   
1  2020-08-12T09:00:08Z  UCRuSxVu4iqTK5kCh90ntAgA                총몇명   
2  2020-08-10T09:54:13Z  UCMVC92EOs9yDJG5JS-CMesQ        양팡 YangPang   
3  2020-08-11T15:00:58Z  UCkQCwnkQfgSuPTTnw_Y7v7w      꽈뚜룹 Quaddurup   
4  2020-08-11T09:00:13Z  UCaO6TYtlC8U5ttz62hTrZgg  JYP Entertainment   

   categoryId         trending_date  \
0          24  2020-08-12T00:00:00Z   
1           1  2020-08-12T00:00:00Z   
2          22  2020-08-12T00:00:00Z   
3          24  2020-08-12T00:00:00Z   
4          10  2020-08-12T00:00:00Z   

                                                tags  view_count   likes  \
0                                          보겸|bokyem     5947503   53326   
1                총몇명|재밌는 만화|부락토스|루시퍼|총몇명 프리퀄|총몇명 스토리      963384   28244   
2  양팡|양팡유튜브|팡튜브|가족시트콤|양팡가족|양팡가족시트콤|양팡언니|현실남매|현실자매...     2950885   17974   
3  꽈뚜룹|한국여행기|quaddurup|뚜룹이|korea|southkorea|vlog|...     1743374   36893   
4  JYP Entertainment|JYP|J.Y.Park|JYPark|박진영|선미|S...     3433885  353337   

   dislikes  comment_count                                  thumbnail_link  \
0    105756         139946  https://i.ytimg.com/vi/uq5LClQN3cE/default.jpg   
1       494           3339  https://i.ytimg.com/vi/I-ZbZCHsHD0/default.jpg   
2     68898          50688  https://i.ytimg.com/vi/9d7jNUjBoss/default.jpg   
3      1798           8751  https://i.ytimg.com/vi/3pI_L3-sMVg/default.jpg   
4      9763          23405  https://i.ytimg.com/vi/zrsBjYukE8s/default.jpg   

   comments_disabled  ratings_disabled  \
0              False             False   
1              False             False   
2              False             False   
3              False             False   
4              False             False   

                                         description  
0                                                NaN  
1  오늘도 정말 감사드립니다!!총몇명 스튜디오 - 총몇명, 십제곱, 5G민, MOVE혁...  
2                                                NaN  
3                앞으로 좀 더 깔끔한 영상제작 약속 드리겠습니다.늘 감사드립니다  
4  MelOn http://kko.to/TWyXd7zYjSpotify https://s...

유튜브 영상의 제목, 올린 날짜, 체널 제목 등이 나와있음을 알 수 있다. 

### 행과 열의 개수 확인하기

In [8]:
# 행과 열 개수
print(df.shape)
nRow, nCol = df.shape
print(f'There are {nRow} rows and {nCol} columns')

(137554, 16)
There are 137554 rows and 16 columns


### 열이름 16개 모두 출력하기

In [9]:
# 열이름 출력하기
df.columns

Index(['video_id', 'title', 'publishedAt', 'channelId', 'channelTitle',
       'categoryId', 'trending_date', 'tags', 'view_count', 'likes',
       'dislikes', 'comment_count', 'thumbnail_link', 'comments_disabled',
       'ratings_disabled', 'description'],
      dtype='object')

### 요약 정보 확인하기

In [10]:
# 요약 정보 확인하기
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 137554 entries, 0 to 137553
Data columns (total 16 columns):
 #   Column             Non-Null Count   Dtype 
---  ------             --------------   ----- 
 0   video_id           137554 non-null  object
 1   title              137554 non-null  object
 2   publishedAt        137554 non-null  object
 3   channelId          137554 non-null  object
 4   channelTitle       137554 non-null  object
 5   categoryId         137554 non-null  int64 
 6   trending_date      137554 non-null  object
 7   tags               137554 non-null  object
 8   view_count         137554 non-null  int64 
 9   likes              137554 non-null  int64 
 10  dislikes           137554 non-null  int64 
 11  comment_count      137554 non-null  int64 
 12  thumbnail_link     137554 non-null  object
 13  comments_disabled  137554 non-null  bool  
 14  ratings_disabled   137554 non-null  bool  
 15  description        134011 non-null  object
dtypes: bool(2), int64(5)

- description열만 Non-Null의 값이 총 행 수와 같지 않음 => 결측치 있음  
    => 확실한 확인 위해 isnull().sum()을 이용해서 확인하기
- categoryId의 type이 정수로 되어 있음 => 범주형 자료로 카테고리별 영상 수를 확인해볼 수 있음  
    => 이후 cat DataFrame과 합칠 수 있는 연결고리
- publishedAt, trending_date가 object 형태로 나와있음 => 시간순으로 패턴 찾을 때 처리 필요할듯!

In [11]:
# 각 열 별 결측치 수 확인하기
# => description에만 결측지가 있고, 나머지 열에서는 결측치가 없음
# 이 경우 결측치를 다 ""나 " "으로 처리해도 괜찮을 것 같음 (삭제할 필요 없음, 굳이 처리 안해도 됨)
df.isnull().sum()

video_id                0
title                   0
publishedAt             0
channelId               0
channelTitle            0
categoryId              0
trending_date           0
tags                    0
view_count              0
likes                   0
dislikes                0
comment_count           0
thumbnail_link          0
comments_disabled       0
ratings_disabled        0
description          3543
dtype: int64

In [12]:
# 요약정보를 제공할 수 있는 int나 float행만 출력가능(여기는 전부 int)
df.describe()

categoryId    view_count         likes       dislikes  comment_count
count  137554.000000  1.375540e+05  1.375540e+05  137554.000000   1.375540e+05
mean       20.426843  1.618720e+06  7.406458e+04     870.091957   8.841621e+03
std         6.267271  6.166372e+06  4.292023e+05    9655.556540   1.125411e+05
min         1.000000  0.000000e+00  0.000000e+00       0.000000   0.000000e+00
25%        19.000000  3.356400e+05  5.743000e+03       0.000000   5.470000e+02
50%        23.000000  6.521940e+05  1.157800e+04     129.000000   1.215000e+03
75%        24.000000  1.307116e+06  2.634875e+04     350.000000   2.870000e+03
max        29.000000  2.963142e+08  1.646425e+07  879358.000000   6.939302e+06

### 분석과는 앞으로 관련없을 열 'thumbnail_link' 제거하기

In [13]:
# 분석과는 앞으로 관련없을 열 'thumbnail_link' 제거하기
df = df.drop('thumbnail_link',axis=1)

### 'title'의 고유한 개수 찾기

In [14]:
# 'title'의 고유한 개수 찾기
df['title'].nunique()

18079

In [15]:
df['title'].unique()

array(['안녕하세요 보겸입니다', '부락토스의 계획 [총몇명 프리퀄]', '평생 반성하면서 살겠습니다.', ...,
       '[EN] ★특별편★ 야구장 스페셜 [둘이 왔어요 EP56]', '어서와~ 한국은 처음이지!?😎🇰🇷',
       '채굴 끝난 줄 알고 무너진 채 방치된 폐광 아래 손도 대지 않은 황금 발견! 금광으로 효용가치는?ㅣ황금빛 꿈을 캐다! 금광 갱도 보수 작업 현장ㅣ극한직업ㅣ#골라듄다큐'],
      dtype=object)

총 영상의 수는 137554개인 것에 반해 'title'의 고유값들은 18079개이다. 이 말은 같은 title을 가진 영상들이 여러개 있다는 것이다. 예를 들어 2020년에 유튜브에서 죄송합니다 챌린지가 있었는데 이때 '죄송합니다'라는 제목으로 여러 영상들이 올라오곤 했다. 이처럼 다른 영상이더라도 제목만 같으면 하나로 묶이기 때문에 갯수가 많이 차이나는 것 같다.

### 'video_id'의 고유한 개수 찾기

In [16]:
# 'video_id'의 고유한 개수 찾기
df['video_id'].nunique()

16930

이 결과를 통해서 데이터셋에 담긴 시기에 영상을 올린 계정의 수를 확인할 수 있다.  
'title'의 경우는 같은 이름을 한 영상을 1개로 count하고, 'video_id'의 경우는 같은 아이디로 만든 영상을 1개로 취급하기 때문에 당연히 다를 수밖에 없다.

### ‘comments_disabled’열이 True라면, ‘comment_count’도 0일 것임. 이것이 사실인지 확인해보기.

In [18]:
df.loc[df['comments_disabled']==True, 'comment_count'].head()

33     0
47     0
191    0
205    0
349    0
Name: comment_count, dtype: int64

###  'ratings_disabled’열이 True라면, ‘likes’, ‘dislikes’ 모두 0일 것임. 이것 또한 사실인지 확인해보기.


In [19]:
df.loc[df['ratings_disabled']==True,['likes', 'dislikes']].head()

likes  dislikes
57       0         0
107      0         0
214      0         0
257      0         0
366      0         0

### 열 ‘publishedAt’, 열 ‘trending_date’ 둘 다 object 형태임을 위에서 확인하였으니, 이를 모두 datatime64[ns] 형태로 변환하기. 

In [20]:
df[['publishedAt', 'trending_date']].head()

publishedAt         trending_date
0  2020-08-09T09:32:48Z  2020-08-12T00:00:00Z
1  2020-08-12T09:00:08Z  2020-08-12T00:00:00Z
2  2020-08-10T09:54:13Z  2020-08-12T00:00:00Z
3  2020-08-11T15:00:58Z  2020-08-12T00:00:00Z
4  2020-08-11T09:00:13Z  2020-08-12T00:00:00Z

In [21]:
df['publishedAt'] = pd.to_datetime(df['publishedAt'], format='%Y-%m-%dT%H:%M:%SZ')

In [22]:
df['trending_date'] = pd.to_datetime(df['trending_date'], format='%Y-%m-%dT00:00:00Z')

In [23]:
print(df[['publishedAt', 'trending_date']].head())

print('-'*30)

print(df[['publishedAt', 'trending_date']].info())

          publishedAt trending_date
0 2020-08-09 09:32:48    2020-08-12
1 2020-08-12 09:00:08    2020-08-12
2 2020-08-10 09:54:13    2020-08-12
3 2020-08-11 15:00:58    2020-08-12
4 2020-08-11 09:00:13    2020-08-12
------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 137554 entries, 0 to 137553
Data columns (total 2 columns):
 #   Column         Non-Null Count   Dtype         
---  ------         --------------   -----         
 0   publishedAt    137554 non-null  datetime64[ns]
 1   trending_date  137554 non-null  datetime64[ns]
dtypes: datetime64[ns](2)
memory usage: 2.1 MB
None


### 열 ‘categoryId’를 int type이 아닌 category type로 변환하기.

In [24]:
df['categoryId'].head()

0    24
1     1
2    22
3    24
4    10
Name: categoryId, dtype: int64

In [25]:
df['categoryId'] = df['categoryId'].astype('category')

In [26]:
df['categoryId'].head()

0    24
1     1
2    22
3    24
4    10
Name: categoryId, dtype: category
Categories (15, int64): [1, 2, 10, 15, ..., 26, 27, 28, 29]

## cat, df의 categoryId와 관련한 문제 해결해보기!

In [27]:
# 각 값이 나온 횟수 출력
print(df['categoryId'].value_counts())

24    47880
22    21699
10    14787
17    10544
23     8626
25     6795
26     5972
20     5085
1      4476
15     2793
27     2691
28     2293
19     1966
2      1708
29      239
Name: categoryId, dtype: int64


카테고리의 경우 24번이 가장 많이 나오고 있습니다(24번이 무엇인지 알아볼 필요 생김)

### cat과 df 합치기

In [28]:
print(df.head(1))
print('-'*30)
print(cat.head(1))

      video_id        title         publishedAt                 channelId  \
0  uq5LClQN3cE  안녕하세요 보겸입니다 2020-08-09 09:32:48  UCu9BCtGIEr73LXZsKmoujKw   

  channelTitle categoryId trending_date       tags  view_count  likes  \
0        보겸 BK         24    2020-08-12  보겸|bokyem     5947503  53326   

   dislikes  comment_count  comments_disabled  ratings_disabled description  
0    105756         139946              False             False         NaN  
------------------------------
                    kind                         etag  id     snippet.title  \
0  youtube#videoCategory  IfWa37JGcqZs-jZeAyFGkbeh6bc   1  Film & Animation   

   snippet.assignable         snippet.channelId  
0                True  UCBR8-60-B28hp2BmDPdntcQ  


In [29]:
# df와 cat의 열이름 맞춰주기 위해
cat.rename(columns={'id':'categoryId'},inplace=True)

In [30]:
# 병합하기
df = pd.merge(df, cat[['categoryId', 'snippet.title']], left_on='categoryId', right_on='categoryId', how='left')

In [31]:
df.head()

video_id                                             title  \
0  uq5LClQN3cE                                       안녕하세요 보겸입니다   
1  I-ZbZCHsHD0                                부락토스의 계획 [총몇명 프리퀄]   
2  9d7jNUjBoss                                   평생 반성하면서 살겠습니다.   
3  3pI_L3-sMVg                              안녕하세요 꽈뚜룹입니다.   
4  zrsBjYukE8s  박진영 (J.Y. Park) When We Disco (Duet with 선미) M/V   

          publishedAt                 channelId       channelTitle  \
0 2020-08-09 09:32:48  UCu9BCtGIEr73LXZsKmoujKw              보겸 BK   
1 2020-08-12 09:00:08  UCRuSxVu4iqTK5kCh90ntAgA                총몇명   
2 2020-08-10 09:54:13  UCMVC92EOs9yDJG5JS-CMesQ        양팡 YangPang   
3 2020-08-11 15:00:58  UCkQCwnkQfgSuPTTnw_Y7v7w      꽈뚜룹 Quaddurup   
4 2020-08-11 09:00:13  UCaO6TYtlC8U5ttz62hTrZgg  JYP Entertainment   

   categoryId trending_date  \
0          24    2020-08-12   
1           1    2020-08-12   
2          22    2020-08-12   
3          24    2020-08-12   
4          10    2020-08-12   

                                                tags  view_count   likes  \
0                                          보겸|bokyem     5947503   53326   
1                총몇명|재밌는 만화|부락토스|루시퍼|총몇명 프리퀄|총몇명 스토리      963384   28244   
2  양팡|양팡유튜브|팡튜브|가족시트콤|양팡가족|양팡가족시트콤|양팡언니|현실남매|현실자매...     2950885   17974   
3  꽈뚜룹|한국여행기|quaddurup|뚜룹이|korea|southkorea|vlog|...     1743374   36893   
4  JYP Entertainment|JYP|J.Y.Park|JYPark|박진영|선미|S...     3433885  353337   

   dislikes  comment_count  comments_disabled  ratings_disabled  \
0    105756         139946              False             False   
1       494           3339              False             False   
2     68898          50688              False             False   
3      1798           8751              False             False   
4      9763          23405              False             False   

                                         description     snippet.title  
0                                                NaN     Entertainment  
1  오늘도 정말 감사드립니다!!총몇명 스튜디오 - 총몇명, 십제곱, 5G민, MOVE혁...  Film & Animation  
2                                                NaN    People & Blogs  
3                앞으로 좀 더 깔끔한 영상제작 약속 드리겠습니다.늘 감사드립니다     Entertainment  
4  MelOn http://kko.to/TWyXd7zYjSpotify https://s...             Music

In [32]:
df.rename(columns={'snippet.title':'category_name'},inplace=True)

In [33]:
df.shape

(137554, 16)

### df['category_name']의 결측값 확인

In [34]:
# category_name의 고유한 값들 확인하기
df['category_name'].unique()

array(['Entertainment', 'Film & Animation', 'People & Blogs', 'Music',
       'Comedy', 'Education', 'News & Politics', 'Sports', nan, 'Gaming',
       'Travel & Events', 'Pets & Animals', 'Science & Technology',
       'Howto & Style', 'Autos & Vehicles'], dtype=object)

In [35]:
# categoryId가 29인 경우의 category_name 확인하기
df.loc[df['categoryId']==29, 'category_name']

17        NaN
150       NaN
303       NaN
476       NaN
640       NaN
         ... 
133281    NaN
133499    NaN
133700    NaN
133938    NaN
134151    NaN
Name: category_name, Length: 239, dtype: object

In [36]:
# 열 ‘category_name’이 결측값인 행의 title 출력해보고 내용 확인하기
print(df['title'][df['category_name'].isnull()])
print('-'*30)
print(df['title'][df['category_name'].isnull()].unique())

17        폭우로 인한 침수 상황시 대처 방법은? [행정안전부X사물궁이]
150       폭우로 인한 침수 상황시 대처 방법은? [행정안전부X사물궁이]
303       폭우로 인한 침수 상황시 대처 방법은? [행정안전부X사물궁이]
476       폭우로 인한 침수 상황시 대처 방법은? [행정안전부X사물궁이]
640       폭우로 인한 침수 상황시 대처 방법은? [행정안전부X사물궁이]
                         ...                
133281                          전한길 x 홍보맨 2편
133499                          전한길 x 홍보맨 2편
133700                          전한길 x 홍보맨 2편
133938                          전한길 x 홍보맨 2편
134151                          전한길 x 홍보맨 2편
Name: title, Length: 239, dtype: object
------------------------------
['폭우로 인한 침수 상황시 대처 방법은? [행정안전부X사물궁이]' '동선 숨겼다가 고발을 당했다. 2억이 나왔다ㅣ넋 나간 가족'
 '[김광일의 입] 文 지지층, “니가 나한테 이럴 수 있어?”' '제가 전생에 무슨 죄가 있어 이 고생을 할까요?'
 '서울시청 가서 세리머니 하는 홍보맨ㅣ충주사과를 찾아라 서울편'
 '6.25전쟁 70주년 기념 유엔참전용사 감사영상 - Thanks to UN veterans.' '도경수 병장 30문 30답'
 '페트병 색깔이 투명해지고 있다고? ㅣ환경부 x 사물궁이' '사막마을 벼수확(5번째 이야기) / 참 까지 얻어먹고~~'
 '사막마을 벼 수확(6번째 이야기).  벼 탈곡기'
 '2020년 제야의 종 온라인 타종행사 | 보신각 현장 행사는 진행하지 않습니다.'
 '2020년 제야의 종 온라인 타종행사 | 2021년 새해맞이 | 보신

TV방송 하이라이트 영상이나 공공기관 홍보 영상으로 보이는 제목이 많이 보인다.

### 열 category_name’에서 결측값을 ‘Nonprofits & Activism’으로 채우기

In [37]:
df.category_name.fillna('Nonprofits & Activism', inplace=True)

In [38]:
# 카테고리 이름 확인하기
df.category_name.unique()

array(['Entertainment', 'Film & Animation', 'People & Blogs', 'Music',
       'Comedy', 'Education', 'News & Politics', 'Sports',
       'Nonprofits & Activism', 'Gaming', 'Travel & Events',
       'Pets & Animals', 'Science & Technology', 'Howto & Style',
       'Autos & Vehicles'], dtype=object)

In [39]:
# 결측값 없음 확인하기
df.category_name.isna().any()

False

In [40]:
df.to_csv('C:/Users/김지희/DSL/dataset/youtube_0716_지희.csv',sep=',')

-----------------------
필수 제출 내용 완료  

# 추가 분석 내용

In [41]:
df.describe()

categoryId    view_count         likes       dislikes  comment_count
count  137554.000000  1.375540e+05  1.375540e+05  137554.000000   1.375540e+05
mean       20.426843  1.618720e+06  7.406458e+04     870.091957   8.841621e+03
std         6.267271  6.166372e+06  4.292023e+05    9655.556540   1.125411e+05
min         1.000000  0.000000e+00  0.000000e+00       0.000000   0.000000e+00
25%        19.000000  3.356400e+05  5.743000e+03       0.000000   5.470000e+02
50%        23.000000  6.521940e+05  1.157800e+04     129.000000   1.215000e+03
75%        24.000000  1.307116e+06  2.634875e+04     350.000000   2.870000e+03
max        29.000000  2.963142e+08  1.646425e+07  879358.000000   6.939302e+06

아무래도 유튜브 트렌딩 데이터셋이어서 

In [42]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 137554 entries, 0 to 137553
Data columns (total 16 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   video_id           137554 non-null  object        
 1   title              137554 non-null  object        
 2   publishedAt        137554 non-null  datetime64[ns]
 3   channelId          137554 non-null  object        
 4   channelTitle       137554 non-null  object        
 5   categoryId         137554 non-null  int64         
 6   trending_date      137554 non-null  datetime64[ns]
 7   tags               137554 non-null  object        
 8   view_count         137554 non-null  int64         
 9   likes              137554 non-null  int64         
 10  dislikes           137554 non-null  int64         
 11  comment_count      137554 non-null  int64         
 12  comments_disabled  137554 non-null  bool          
 13  ratings_disabled   137554 non-null  bool    

In [43]:
# trend dataset임에도 불구하고, 조회수가 0개로 나와있어서, 측정이 잘못된 것인지 확인해보려고 함.
df.iloc[df['view_count'].idxmin()]

video_id                                                   gNfxYzW-wME
title                      Hello 2021 Korea | 크리에이터와 함께하는 Goodbye 2020
publishedAt                                        2020-12-31 14:00:10
channelId                                     UCOH52Yqq4-rdLvpt2Unsqsw
channelTitle                                             YouTube Korea
categoryId                                                          24
trending_date                                      2021-01-01 00:00:00
tags                 YouTube Originals|YouTube|Hello 2021 Korea|ful...
view_count                                                           0
likes                                                             2052
dislikes                                                            63
comment_count                                                      627
comments_disabled                                                False
ratings_disabled                                                 False
descri

좋아요도 좀 달리고, 댓글도 달린 것에 비해 조회수가 한개도 없는 것으로 나왔다. 이상하다..

In [44]:
# 최고 조회수 영상
df.iloc[df['view_count'].idxmax()]

video_id                                                   WMweEpGlu_U
title                                 BTS (방탄소년단) 'Butter' Official MV
publishedAt                                        2021-05-21 03:46:13
channelId                                     UC3IZKseVpdzPSBaWxBxundA
channelTitle                                               HYBE LABELS
categoryId                                                          10
trending_date                                      2021-06-04 00:00:00
tags                                   BIGHIT|빅히트|방탄소년단|BTS|BANGTAN|방탄
view_count                                                   296314174
likes                                                         16464253
dislikes                                                        181350
comment_count                                                  6939302
comments_disabled                                                False
ratings_disabled                                                 False
descri

In [47]:
df.iloc[df['dislikes'].idxmax()]

video_id                                                   vRXZj0DzXIA
title                  BLACKPINK - 'Ice Cream (with Selena Gomez)' M/V
publishedAt                                        2020-08-28 04:00:11
channelId                                     UCOmHUn--16B90oW2L6FRR3A
channelTitle                                                 BLACKPINK
categoryId                                                          10
trending_date                                      2020-09-05 00:00:00
tags                 YG Entertainment|YG|와이지|K-pop|BLACKPINK|블랙핑크|블...
view_count                                                   184778248
likes                                                         11795696
dislikes                                                        879358
comment_count                                                  2735999
comments_disabled                                                False
ratings_disabled                                                 False
descri

In [48]:
df.plot.scatter(x='dislikes'.
                  y='biew_count', 
                  s=50, # marker size
                  c='blue', 
                  alpha=0.5)
plt.title('Scatter Plot of iris by pandas', fontsize=20)
plt.xlabel('Petal Length', fontsize=14)
plt.ylabel('Petal Width', fontsize=14)
plt.show()

SyntaxError: invalid character in identifier (2891516615.py, line 2)

In [51]:
print(df['trending_date'].value_counts().head(40))

KeyError: False

In [67]:

print(df['channelTitle'].value_counts())

짤툰                         792
파뿌리                        715
SPOTV                      690
총몇명                        684
BANGTANTV                  648
                          ... 
LOTTEChemical Official       1
초아시티                         1
대륙남TV [clark tv]             1
다키포스트DAKI POST               1
東映アニメーション公式YouTubeチャンネル      1
Name: channelTitle, Length: 3039, dtype: int64


In [79]:
df[df['title'] in 'Butter']

TypeError: 'in <string>' requires string as left operand, not Series

In [53]:
df.query('trending_date'=='2021-05-24')['category_name']

ValueError: expr must be a string to be evaluated, <class 'bool'> given